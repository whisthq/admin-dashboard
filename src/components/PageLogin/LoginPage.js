import React, { Component } from 'react'
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'
import Table from 'react-bootstrap/Table'
import Container from 'react-bootstrap/Container'
import FormControl from 'react-bootstrap/FormControl'
import InputGroup from 'react-bootstrap/InputGroup'
import Popup from "reactjs-popup";
import '../../static/App.css';

import Button from 'react-bootstrap/Button'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux';
import { HashLink } from 'react-router-hash-link';
import { ReactTypeformEmbed } from 'react-typeform-embed';
import { loginUser } from '../../actions/index.js'

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faEdit, faCheck, faCircleNotch, faTimes } from '@fortawesome/free-solid-svg-icons'


class LoginPage extends Component {
  constructor(props) {
    super(props)
    this.state = {emailLogin: '', passwordLogin: '', loggingIn: false, failed_login_attempt: false}
    this.updateWindowDimensions = this.updateWindowDimensions.bind(this)
  }

  componentDidMount() {
    this.updateWindowDimensions()
    window.addEventListener('resize', this.updateWindowDimensions)
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.updateWindowDimensions)
  }

  updateWindowDimensions() {
    this.setState({ width: window.innerWidth, height: window.innerHeight })
  }

  changeEmailLogin = (evt) => {
    this.setState({
      emailLogin: evt.target.value
    });
  }

  loginKeyPress = (event) => {
    if(event.key === 'Enter'){
      this.setState({loggingIn: true, failed_login_attempt: false})
      this.props.dispatch(loginUser(this.state.emailLogin, this.state.passwordLogin));
    }
  }

  handleLogin = () => {
    this.setState({loggingIn: true, failed_login_attempt: false})
    this.props.dispatch(loginUser(this.state.emailLogin, this.state.passwordLogin));
  }

  changePasswordLogin = (evt) => {
    if(evt.key === 'Enter') {
      this.props.dispatch(loginUser(this.state.emailLogin, this.state.passwordLogin));
    } else {
      this.setState({
        passwordLogin: evt.target.value
      });
    }
  }

  componentDidUpdate(prevProps) {
    if(prevProps.login_attempts != this.props.login_attempts && !this.state.failed_login_attempt) {
      this.setState({failed_login_attempt: true, loggingIn: false})
    }
  }

  render() {
    let modalClose = () => this.setState({ modalShow: false })
    if (this.state.width > 700 && this.state.modalShow) {
      modalClose()
    }
    return (
      <div style = {{height: '100vh'}}>
        <div style = {{width: 300, margin: 'auto', marginTop: 150}}>
          <div style = {{textAlign: 'center', fontWeight: 'bold', fontSize: 30, marginBottom: 30}}>
            Fractal Admin
          </div>
          {
          this.state.failed_login_attempt
          ?
          <div style = {{textAlign: 'center', fontSize: 14, color: "#f9000b", background: "#fdf0f1", width: '100%', padding: 10, borderRadius: 5, fontWeight: 'bold', marginBottom: 20}}>
            Invalid credentials
          </div>
          :
          <div></div>
          }
          <InputGroup className="mb-3">
            <FormControl
              type = "email"
              aria-label="Default"
              aria-describedby="inputGroup-sizing-default"
              placeholder = "Username"
              onChange = {this.changeEmailLogin}
              onKeyPress = {this.loginKeyPress}
              style = {{borderRadius: 5, maxWidth: 600, backgroundColor: "#F4F4F4", border: "none", padding: "30px 20px"}}
            /><br/>
          </InputGroup>

          <InputGroup className="mb-3" style = {{marginTop: 20}}>
            <FormControl
              aria-label="Default"
              type = "password"
              aria-describedby="inputGroup-sizing-default"
              placeholder = "Password"
              onChange = {this.changePasswordLogin}
              onKeyPress = {this.loginKeyPress}
              style = {{borderRadius: 5, maxWidth: 600, backgroundColor: "#F4F4F4", border: "none", padding: "30px 20px"}}
            />
          </InputGroup>
          {
          !this.state.loggingIn
          ?
          <Button  onClick = {this.handleLogin} style = {{marginTop: 10, fontWeight: 'bold', color: 'white', width: '100%', border: 'none', background: 'black', boxShadow: '0px 2px 4px rgba(0, 0, 0, 0.25)', padding: '15px'}}>
              LOG IN
          </Button>
          :
          <Button  disabled = "true" style = {{marginTop: 10, fontWeight: 'bold', color: 'white', width: '100%', border: 'none', background: 'black', boxShadow: '0px 2px 4px rgba(0, 0, 0, 0.25)', padding: '15px', textAlign: 'center'}}>
              <FontAwesomeIcon icon = {faCircleNotch} spin style = {{color: 'white', fontSize: 13}}/>
          </Button>
          }
        </div>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    login_attempts: state.AccountReducer.login_attempts
  }
}

export default connect(mapStateToProps)(LoginPage)
