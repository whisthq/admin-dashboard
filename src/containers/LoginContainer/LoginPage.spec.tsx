import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import Button from 'react-bootstrap/Button'
import InputGroup from 'react-bootstrap/InputGroup'

import { Login } from './LoginPage'

// necessary for enzyme to work
Enzyme.configure({ adapter: new Adapter() })

/*
A login page. Should have a login button and two inputs: one for the password and one for the username.
It should correctly update the state and dispatch (for middleware to call server) on login. It should remember
when it's failed and stuff to show you if you have some number of failed login attempts.
 */
describe('<Login />', () => {
    // snapshot test
    it('renders correctly', () => {
        const wrapper = shallow(
            <Login
                login_attempts={0}
                authenticated={false}
                dispatch={(items: any) => {
                    // mock
                }}
            />
        )

        expect(toJson(wrapper)).toMatchSnapshot()
    })

    test('on component update it sets state correctly', () => {
        // TODO
    })
    test('helpers correctly dispatch and change state', () => {
        // TODO (optional) here to encourage you to test the helpers are necessary
    })

    it('displays two inputs and one button regardless of state', () => {
        let wrapper = shallow(
            <Login
                login_attempts={0}
                authenticated={false}
                dispatch={(items: any) => {
                    // mock
                }}
            />
        )

        expect(wrapper.find(InputGroup)).toHaveLength(2)
        expect(wrapper.find(Button)).toHaveLength(1)
    })

    // TODO (stateful) check whether the loading icon properly shows when it's in the loading state
})
