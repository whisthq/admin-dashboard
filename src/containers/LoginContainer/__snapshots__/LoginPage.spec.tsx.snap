// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders correctly 1`] = `
<div
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <div
    style={
      Object {
        "margin": "auto",
        "marginTop": 150,
        "width": 300,
      }
    }
  >
    <div
      style={
        Object {
          "fontSize": 30,
          "fontWeight": "bold",
          "marginBottom": 30,
          "textAlign": "center",
        }
      }
    >
      Fractal Admin
    </div>
    <div />
    <InputGroup
      className="mb-3"
    >
      <FormControl
        aria-describedby="inputGroup-sizing-default"
        aria-label="Default"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="Username"
        style={
          Object {
            "backgroundColor": "#e3e5e8",
            "border": "none",
            "borderRadius": 5,
            "maxWidth": 600,
            "padding": "30px 20px",
          }
        }
        type="email"
      />
      <br />
    </InputGroup>
    <InputGroup
      className="mb-3"
      style={
        Object {
          "marginTop": 20,
        }
      }
    >
      <FormControl
        aria-describedby="inputGroup-sizing-default"
        aria-label="Default"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="Password"
        style={
          Object {
            "backgroundColor": "#e3e5e8",
            "border": "none",
            "borderRadius": 5,
            "maxWidth": 600,
            "padding": "30px 20px",
          }
        }
        type="password"
      />
    </InputGroup>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "background": "#111111",
          "border": "none",
          "color": "white",
          "fontWeight": "bold",
          "marginTop": 10,
          "padding": "15px",
          "width": "100%",
        }
      }
      variant="primary"
    >
      LOG IN
    </Button>
  </div>
</div>
`;
